### DISCUSSION TITLE
Is possible to have open popup witk marker ?

### DISCUSSION THREAD

**Discussion by ambarja** [maplibre/maplibre-gl-js#1670] [Q&A] [answered]
Hi everyone, I am recently using MapLibre for my personal projects and I would like to have my marker with a popup open, however it doesn't allow me to do it.
Could someone help me? 

```js
var map = new maplibregl.Map({
    container: 'map',
    style:
    'https://api.maptiler.com/maps/outdoor/style.json?key=KEY',
    center: [-71.04286,-13.85728],
    zoom: 10
    }).addControl(new maplibregl.NavigationControl(), 'top-right');

// New Popup
var popup = new maplibregl.Popup({offset: 25,closeOnClick: false})
  .setHTML('<b> Hi everyone! My name is Antony :3 and I am a Geographer in internet </b>');

// create DOM element for the marker
var el = document.createElement('div');
el.id = 'marker';

var maker = new maplibregl.Marker(el)
  .setLngLat([-71.04286,-13.85728])
  .setPopup(popup)
  .addTo(map);
```
![image](https://user-images.githubusercontent.com/23284899/191654326-f456aa5f-41be-4f59-8f31-b2c59e77bde8.png)

**Comment by HarelM** [comment]
Take a look at the examples docs, here's a similar one:
https://maplibre.org/maplibre-gl-js-docs/example/custom-marker-icons/

**Reply by ambarja** [reply]
> Take a look at the examples docs, here's a similar one: https://maplibre.org/maplibre-gl-js-docs/example/custom-marker-icons/
Yes, but my question is if it is possible to have a popup open with a custom marker. Here is an example of how I would like to have this.
**Observation: Open popup without to do click** 
 - [https://maplibre.org/maplibre-gl-js-docs/example/popup/](https://maplibre.org/maplibre-gl-js-docs/example/popup/)
 - [https://maplibre.org/maplibre-gl-js-docs/example/set-popup/](https://maplibre.org/maplibre-gl-js-docs/example/set-popup/)
![image](https://user-images.githubusercontent.com/23284899/191742528-13137002-27fe-4ebb-8fa7-b8d8905bd211.png)

**Reply by wipfli** [reply]
Nice, if this is a good answer, feel free to press "Mark as answer"...

**Comment by ambarja** [comment, marked as answer]
Thanks, but the answer was the following: 

```js
// Basic basemap
var map = new maplibregl.Map({
    container: 'map',
    style:
    'https://api.maptiler.com/maps/outdoor/style.json?key=hdV4nOYKQa4di4tH4qr2	',
    center: [-71.04286,-13.85728],
    zoom: 10
    }).addControl(new maplibregl.NavigationControl(), 'top-right');

// New Popup
var popup = new maplibregl.Popup({offset: 25,closeOnClick: false})
  .setHTML('<b> Hi everyone! My name is Antony :3 and I am a Geographer in internet </b>')
  .setLngLat([-71.04286,-13.85728])
  .addTo(map);
  
// create DOM element for the marker
var el = document.createElement('div');
el.id = 'marker';
var maker = new maplibregl.Marker(el)
    .setLngLat([-71.04286,-13.85728])
    .setPopup(new maplibregl.Popup({offset: 25}).setHTML('<b> Hi everyone! My name is Antony :3 and I am a Geographer in internet </b>'))
    .addTo(map);
```
